### DISCUSSION TITLE
incorrect $skiptoken when using custom name for key

### DISCUSSION THREAD

**Discussion by ravz** [OData/AspNetCoreOData#960] [Q&A]
Hi, I'm using 8.2.0 OData connected to EF7 SQL database

For the key of one of my tables I'm using a custom name that does not match the column name

```
[Key]
[DataMember(Name="CostId")]
public int Id { get; set; }
```

Everything works fine except the generated skiptoken uri comes out as
`$skiptoken=CostId-1000`
which does not work, if I change it to
`$skiptoken=Id-1000`
it does work

I created a custom skip token handler to specifically correct for this table, but I was wondering if there is a more generic way to fix this and/or is this a bug that I should create an Issue for?

**Comment by habbes** [comment]
Hello @ravz. This sounds like a potential bug. How does your custom skip token handler fix the issue? Does it rewrite the `$skiptoken` such that it uses `Id-1000` or does it get `CostId-1000` to work? Or does it do something else entirely?
